core/src/test/scala/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriterSuite.scala

```diff
@@ -118,6 +124,26 @@ class BypassMergeSortShuffleWriterSuite extends SparkFunSuite with BeforeAndAfte
           blockIdToFileMap.get(invocation.getArguments.head.asInstanceOf[BlockId]).get
         }
     })
+    when(shuffleExecutorComponents.writes()).thenReturn(writeSupport)
+    when(writeSupport.createMapOutputWriter(
```

Comment:
Why not just use a real `DefaultShuffleWriteSupport`? This is an integration test so we can feel free to use real objects if they're on the critical path.